include/zephyr/net/net_config.h

```diff
@@ -14,6 +14,13 @@
 #include <zephyr/types.h>
 #include <zephyr/device.h>
 #include <zephyr/net/net_if.h>
+#include <zephyr/net/ieee802154_mgmt.h>
+
+#if defined(CONFIG_NET_CONFIG_SETTINGS)
+#include <net_init_config.inc>
+#else
+struct net_init_config;
```

Comment:
Is this struct generated from the yaml input? If so, almost any change would result in a different `sizeof` and make it unable to read from the settings subsystem, correct?

I think that's an issue to be able to update your devices.

Reply:
The size would change only if one changes the schema. There is a hash embedded into the data to detect this and discard the saved data if that happens.